assets/src/edit-story/components/form/color/colorPreview.js

```diff
@@ -88,11 +101,78 @@ function ColorPreview({ onChange, hasGradient, hasOpacity, value, label }) {
     hasOpacity,
   ]);
 
+  const [hexInputValue, setHexInputValue] = useState('');
+
+  useEffect(() => setHexInputValue(previewText), [previewText]);
+
+  const colorType = value?.type;
+  const isEditable =
+    !isMultiple &&
+    Boolean(previewText) &&
+    (!colorType || colorType === 'solid');
+
+  const editLabel = __('Edit', 'web-stories');
+  const inputLabel = __('Enter', 'web-stories');
+
+  const buttonProps = {
+    as: 'button',
+    type: 'button', // avoid submitting forms
+    onClick: handleOpenEditing,
+    'aria-label': `${editLabel}: ${label}`,
+  };
+
+  const handleInputChange = useCallback(
+    (evt) => {
+      const raw = evt.target.value.trim();
+      // Strip initial '#' (might very well be pasted in)
+      const val = raw.charAt(0) === '#' ? raw.substr(1) : raw;
+      setHexInputValue(val);
+      const hasNonHex = /[^0-9a-f]/i.test(val);
+      const hasValidLength = val.length === 6;
```

Comment:
`3` is also correct.

Reply:
I had that at first, but it was superannoying that while typing it would expand to a 6-digit hex as soon as you had a 3 digit hex. As mentioned in the PR description above, we could support 3 digit hex on blur only, but that's pretty weird UX.